src/components/topo/ucc_topo.h

```diff
@@ -253,9 +254,10 @@ static inline ucc_rank_t ucc_topo_nnodes(ucc_topo_t *topo)
     return sbgp->group_size;
 }
 
-/* Returns an array mapping each rank to its node leader.
-   The array is cached in topo->node_leaders. */
-ucc_status_t ucc_topo_get_node_leaders(ucc_topo_t *topo,
-                                       ucc_rank_t **node_leaders_out);
+/* Returns node leaders array - array that maps each rank to the TEAM RANK that 
+   is the leader of that rank's node. Also returns per-node leaders array - array
+   mapping node_id to the TEAM RANK of that node's leader */
+ucc_status_t ucc_topo_get_node_leaders(ucc_topo_t *topo, ucc_rank_t **node_leaders_out,
```

Comment:
Here I updated the previous PR that defined this function to also take in an out-argument for the node leaders. The rest of the functionality is the same.